File: Accelbuffer/Accelbuffer/ITypeSerializer.cs
namespace Accelbuffer
{
    /// <summary>
    /// 实现接口完成指定类型对象的序列化和反序列化
    /// </summary>
    /// <typeparam name="T">序列化的对象的类型</typeparam>
    public interface ITypeSerializer<T>
    {
        /// <summary>
        /// 方法用于实现对象的序列化
        /// </summary>
        /// <param name="obj">将被序列化的对象</param>
        /// <param name="writer">数据输出流</param>
        void Serialize(T obj, ref AccelWriter writer);

        /// <summary>
        /// 方法用于实现对象的反序列化
        /// </summary>
        /// <param name="reader">数据输入流</param>
        /// <returns>反序列化对象</returns>
        T Deserialize(ref AccelReader reader);
    }
}

File: Accelbuffer/Accelbuffer.Java/src/accelbuffer/core/ITypeSerializer.java
package accelbuffer.core;

/**
 * 实现接口完成指定类型对象的序列化和反序列化
 * @param <T> 序列化的对象的类型
 */
public interface ITypeSerializer<T> {
    /**
     * 方法用于实现对象的序列化
     * @param obj 将被序列化的对象
     * @param writer 数据输出流
     */
    public void serialize(T obj, AccelWriter writer);

    /**
     * 方法用于实现对象的反序列化
     * @param reader 数据输入流
     * @return 反序列化对象
     */
    public T deserialize(AccelReader reader);
}
